pkg/schedule/checker/affinity_checker.go

```diff
@@ -103,24 +103,27 @@ func (c *AffinityChecker) Check(region *core.RegionInfo) []*operator.Operator {
 		// Region doesn't belong to any affinity group
 		return nil
 	}
-	// If the Group is not affinity scheduling allowed, provide the available Region information and fetch it again.
-	if !group.IsAffinitySchedulingAllowed {
+
+	// If the Group is not affinity scheduling allowed, provide the available Region information and fetch group state again.
+	needRefetch := !group.IsAffinitySchedulingAllowed
+
+	// If Region is affinity, but the Group is not replicated, expire the Group.
+	// Then provide the available Region information and fetch the Group state again.
+	if isAffinity && c.isGroupReplicated(region, group) {
```

Comment:
```suggestion
	if isAffinity && !c.isGroupReplicated(region, group) {
```
**but the Group is not replicated**